app/models/line_item.rb

```diff
@@ -5,9 +5,13 @@ class LineItem < ApplicationRecord
   belongs_to :product
   belongs_to :cart, optional: true
 
+  validates :product, uniqueness: { 
+    scope: :cart,
```

Comment:
what if I have used cart_id instead of cart?

Reply:
Since **cart** is created as association in LineItem, rails will automatically **cart_id** into account when using validation

However, if **cart_id** is used then we are explicitly specifying the attribute that should be unique when scoped with **product**. Hence no such difference here if **cart** or **cart_id** used. But if **cart** association removed then using **cart** will throw error but **cart_id** will not.